lib/kubeclient/watch_stream.rb

```diff
@@ -36,6 +32,19 @@ def each
         raise unless @finished
       end
 
+      def generate_request
+        request = Net::HTTP::Get.new(@uri)
+        if @options[:basic_auth_user] && @options[:basic_auth_password]
+          request.basic_auth @options[:basic_auth_user],
+                             @options[:basic_auth_password]
+        end
+
+        if @options[:bearer_token]
```

Comment:
does this imply you might have both basic_auth and bearer token at the same time?